core/capabilities/triggers/logevent/trigger.go

```diff
@@ -0,0 +1,196 @@
+package logevent
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+// Log Event Trigger Capability Request Config Details
+type RequestConfig struct {
+	ContractName      string `json:"contractName"`
+	ContractAddress   string `json:"contractAddress"`
+	ContractEventName string `json:"contractEventName"`
+	// Log Event Trigger capability takes in a []byte as ContractReaderConfig
+	// to not depend on evm ChainReaderConfig type and be chain agnostic
+	ContractReaderConfig map[string]any `json:"contractReaderConfig"`
+}
+
+// LogEventTrigger struct to listen for Contract events using ContractReader gRPC client
+// in a loop with a periodic delay of pollPeriod milliseconds, which is specified in
+// the job spec
+type logEventTrigger struct {
+	ch   chan<- capabilities.TriggerResponse
+	lggr logger.Logger
+
+	// Contract address and Event Signature to monitor for
+	reqConfig      *RequestConfig
+	contractReader types.ContractReader
+	relayer        core.Relayer
+	startBlockNum  uint64
+
+	// Log Event Trigger config with pollPeriod and lookbackBlocks
+	logEventConfig Config
+	ticker         *time.Ticker
+	cancel         context.CancelFunc
```

Comment:
```suggestion
```

Reply:
cancel is also called in the Stop function, which is called from UnregisterTrigger, at which point we do not have access to the trigger's cancel function in the caller. So this struct member is needed

Reply:
Which usages of the corresponding `context.Context` can still be in flight when `Stop` is called?